xbmc/music/MusicDatabase.cpp

```diff
@@ -8888,6 +8951,100 @@ bool CMusicDatabase::GetGenresJSON(CFileItemList& items, bool bSources)
   return false;
 }
 
+bool CMusicDatabase::GetBoxsetDiscs(const std::string& strBaseDir, CFileItemList& items, int idAlbum, const SortDescription &sortDescription)
+{
+  // Get album details and fill item list with copy for each disc in box set 
+  // adding disc number and subtitle per item
+  // ! @TODO: Convert into GetDiscsByWhere,  handle no idAlbum => all discs
+  // 1 disc => same as album, is acceptable. Maybe use JOIN rather than 2 queries?
+
```

Comment:
`GetBoxsetDiscs` should be renamed `GetDiscsByWhere` as used for any multi-disc album not just boxed sets. 
Also consider effect of other filter/options e.g. say showing just albums that have songs of genre = X (and nothing says all the songs on an album have the same genre), or have an artist filter and showing those albums with songs by an artists including guest appearences and on compilations. Navigation to discs should just show those discs with songs that staify the criteria. Look at how GetAlbumsByWhere uses GetFilter to do that for hints (or ask)

Reply:
Do you mean, if an album has four discs and three discs contain tracks of genre 'pop' and one disc tracks of 'rock', you want to just display either the three discs or the one, depending upon the filter ?? And similar for an artist or any other filter criteria ?

Reply:
Sorry for leaving you hanging, my limited Kodi time has been eaten by other things.
Yes, to make the navigation similar to other nodes (but may need some consideration what behaviour is wanted). Artists is a better example: say you pick an artist from "artists" node, albums are listsed including an album that is a boxed set where the artist makes a guest appearance on 2 songs on different discs. When you click on that album what is displayed - discs or songs? You could say just a few song go directly to them, but we don't skip the album level when the artist only has one album so should we skip discs?

Reply:
We all have lives and limited time, so no worries on that score. I won't have much time myself over the next three days or so....
Anyway,  I have thought about this quite a lot.  I'm not convinced myself that it's actually needed, however that's probably for a different day.  In the meantime, from an implementation point of view, GetFilter() currently only understands a type of 'albums'  'artists' or 'songs'.  I have contemplated adding a 'discs' type (which might come in useful further along the line ??) but haven't done so yet because basically we are actually filtering the songs that are on a particular disc.  Discs themselves don't have artists or genres or roles and there is no structure to link them to such things.
With that in mind, I have been playing with making GetFilter always use a type of 'songs' if 'disc' is present as an option in the url.  This makes it include 'discid' and 'disctitle' (both of which are song options anyway) and then apply any other filter rules (for songs) in the same way they get applied for other routines that call it. 

If I don't do it like that, GetFilter sees a type of 'albums' but trying to filter discs by album doesn't make sense as you have to have an album first before you can get any discs for it.
I'm still not convinced that I want to be able to return say 15 individual discs that have an artist of 'blah' and a genre of 'foo'.  If I get back discs with no actual titles (the most common), I could end up with a list of 'disc 1, disc 1, disc 3, disc 2, disc 1, disc 1, etc' .

Reply:
OK, I'll see if I can give more clarity on what I had in mind here too. But that will be later over the weekend :)